### PR TITLE
Port all tests to zio-test

### LINKED ISSUE #12: Move to using zio-test

### PR DESCRIPTION
**Pull request by mijicd**
Closes #12

### CODE REVIEW

File: zio-config/src/test/scala/zio/config/ListAndOptionalTest.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  final case class OverallConfig(option: Option[Option[Option[Id]]])
+
+  val genOverallConfig =
+    Gen.option(genId).map(t => OverallConfig(t.map(t => Option(Option(t)))))
+
+  private val cId = string("kId").xmap(Id)(_.value)
+
+  val cOverallConfig =
+    cId.optional.optional.optional.xmap(OverallConfig)(_.option)
+}
```
Comment (afsalthaj) [inline zio-config/src/test/scala/zio/config/ListAndOptionalTest.scala]: Nice , looking at the consistency  between `Test extends BaseSpec` and `TestUtils` everywhere.
Much cleaner now

**Comment by afsalthaj** [comment]
@mijicd Thanks for this PR 👍 

Could you resolve conflicts and merge

**Comment by afsalthaj** [comment]
If possible wait for my PR to get merged.

**Comment by mijicd** [comment]
@afsalthaj synced with master, but you'll need to re-approve it eventually :). Which PR should I wait, #78 ?

**Comment by afsalthaj** [comment]
@mijicd approved again. Yes #78 .

**Comment by mijicd** [comment]
No problems, I'll sync & merge this one as soon as yours is in.

**Comment by afsalthaj** [comment]
I know it’s a bit of change. It’s mostly around not having to deal with tuple
When reading and lifting to zio when writing.

**Comment by mijicd** [comment]
@afsalthaj it's kinda more involving than the PR itself :)

**Comment by mijicd** [comment]
@afsalthaj synced

**Pull request merged**